nemo_skills/pipeline/utils.py

```diff
@@ -758,6 +848,52 @@ def get_cluster_config(cluster=None, config_dir=None):
     return cluster_config
 
 
+def update_ssh_tunnel_config(cluster_config: dict):
+    """
+    Update the ssh tunnel configuration in the cluster config to resolve job dir and username.
+    uses the `user` information to populate `job_dir` if in config.
+
+    Args:
+        cluster_config: dict: The cluster configuration dictionary
+
+    Returns:
+        dict: The updated cluster configuration dictionary
+    """
+    if 'ssh_tunnel' not in cluster_config:
+        return cluster_config
+
+    if 'user' in cluster_config['ssh_tunnel']:
+        # Resolve `user` from env if not provided
+        if cluster_config['ssh_tunnel']['user'] is None:
+            cluster_config['ssh_tunnel']['user'] = os.environ['USER']
+            LOG.info(f"Resolved `user` to `{cluster_config['ssh_tunnel']['user']}`")
+
+        elif isinstance(cluster_config['ssh_tunnel']['user'], str) and '$' in cluster_config['ssh_tunnel']['user']:
+            cluster_config['ssh_tunnel']['user'] = os.path.expandvars(cluster_config['ssh_tunnel']['user'])
```

Comment:
can we just have some recursive logic to do this for all variables, not just ssh tunnel? That would make things more consistent and maybe there is already some library we can reuse that does that. But if it's hard, ok to keep it like this

Reply:
and if not, then at least this function can be much shorter by having a for loop over all variables and same logic inside the loop, instead of current copy-paste

Reply:
Done